packages/pieces/community/pinterest/src/lib/actions/update-board.ts

```diff
@@ -0,0 +1,45 @@
+import { createAction, Property } from '@activepieces/pieces-framework';
+import { makeRequest } from '../common';
+import { pinterestAuth } from '../common/auth';
+import { HttpMethod } from '@activepieces/pieces-common';
+import { boardIdDropdown } from '../common/props';
+
+
+export const updateBoard = createAction({
+  auth: pinterestAuth,
+  name: 'updateBoard',
+  displayName: 'Update Board',
+  description: 'Modify a board’s name, description, or visibility settings.',
+  props: {
+    board_id: boardIdDropdown,
+    name: Property.ShortText({
+      displayName: 'Board Name',
+      required: true,
+      description: 'The new name of the board.'
+    }),
+    description: Property.LongText({
+      displayName: 'Description',
+      required: true,
+      description: 'The new description of the board.'
+    }),
+    privacy: Property.StaticDropdown({
+      displayName: 'Privacy',
+      required: true,
+      options: {
+        options: [
+          { label: 'Public', value: 'PUBLIC' },
+          { label: 'Secret', value: 'SECRET' }
+        ]
+      },
+      description: 'Set the board as public or secret.'
+    })
+  },
+  async run({ auth, propsValue }) {
```

Comment:
Inefficient Body Building: The current logic always builds body properties even when not needed:
```suggestion
     async run({ auth, propsValue }) {
     const { board_id, name, description, privacy, ad_account_id } = propsValue;
     
     // Only include fields that are actually being updated
     const body: any = {};
     if (name && name.trim()) body.name = name.trim();
     if (description !== undefined) body.description = description; // Allow empty string
     if (privacy) body.privacy = privacy; // Remove toLowerCase()
     
     let path = `/boards/${board_id}`;
     if (ad_account_id) {
       path = `/boards/${board_id}?ad_account_id=${ad_account_id}`;
     }
     
     return await makeRequest(auth.access_token as string, HttpMethod.PATCH, path, body);
   }
```